src/hotspot/share/classfile/protectionDomainCache.cpp

```diff
@@ -35,23 +35,24 @@
 #include "oops/oop.inline.hpp"
 #include "oops/weakHandle.inline.hpp"
 #include "runtime/atomic.hpp"
+#include "runtime/mutexLocker.hpp"
 #include "utilities/growableArray.hpp"
-#include "utilities/hashtable.inline.hpp"
+#include "utilities/resourceHash.hpp"
 
-unsigned int ProtectionDomainCacheTable::compute_hash(Handle protection_domain) {
-  // Identity hash can safepoint, so keep protection domain in a Handle.
-  return (unsigned int)(protection_domain->identity_hash());
+unsigned int ProtectionDomainCacheTable::compute_hash(const WeakHandle& protection_domain) {
```

Comment:
We are we now using `WeakHandle` everywhere?

Reply:
WeakHandle is the object we're storing as in the hashtable.  It also turns out to be the key.